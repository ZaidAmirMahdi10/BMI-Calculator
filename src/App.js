// src/App.js
import React from 'react';
import logo from './logo.svg';
import './App.css';
import BMICalculator from './BMICalculator';

function App() {
  return (
    <div className="App">
      <header className="App-header">
        <BMICalculator />
      </header>
    </div>
  );
}

export default App;
