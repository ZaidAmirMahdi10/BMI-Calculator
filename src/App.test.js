// src/App.test.js

import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom'
import App from './App';

test('renders BMI Calculator header', () => {
  render(<App />);
  const headerElement = screen.getByText(/BMI Calculator/i);
  expect(headerElement).toBeInTheDocument();
  expect(headerElement.tagName).toBe('H1'); 
});

test('matches snapshot', () => {
  const { asFragment } = render(<App />);
  expect(asFragment()).toMatchSnapshot();
});
