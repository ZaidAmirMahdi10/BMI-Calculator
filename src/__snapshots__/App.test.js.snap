// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <header
      class="App-header"
    >
      <div>
        <h1>
          BMI Calculator
        </h1>
        <div>
          <label>
            Height (cm):
            <input
              type="number"
              value=""
            />
          </label>
        </div>
        <div>
          <label>
            Weight (kg):
            <input
              type="number"
              value=""
            />
          </label>
        </div>
        <button>
          Calculate BMI
        </button>
      </div>
    </header>
  </div>
</DocumentFragment>
`;
